CMake hacks for MCST e2k (Elbrus 2000) architecture

r-a-sattarov opened the pull request:
- luajit.cmake: added hack for MCST version of LuaJIT
- xrGame/CMakeLists: added hack for MCST lcc compiler

gshep committed:
luajit.cmake: added hack for MCST version of LuaJIT
Used pre-generated buildvm_arch.h when compiling LuaJIT from MCST

Masterkatze commented on src/xrGame/CMakeLists.txt:
```diff
@@ -2673,6 +2673,13 @@ target_compile_options(${PROJECT_NAME}
     -Wno-trigraphs
 )
 
+if ((CMAKE_BUILD_TYPE STREQUAL "Debug" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo") AND PROJECT_PLATFORM_E2K) # E2K: hack for debug compile (MCST lcc compiler does not support a file > 4Gb, this is a bug in EDG frontend)
```
Use generator expression please

a1batross replied:
What's the purpose of using generator expression here?

I would replace the debug check to something else though, but I forgot CMake (as a scary nightmare)

Masterkatze replied:
The purpose is that we don't need to have multiple `target_compile_options` for single target.

In result, it will look like this (needs testing):
```
target_compile_options(${PROJECT_NAME} 
    PRIVATE
    -Wno-trigraphs
    $<$<AND:$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>,$<BOOL:PROJECT_PLATFORM_E2K>>:-g0>
)
```

Xottab-DUTY replied:
@Masterkatze, it looks unreadable. Readability is preferred.

Xottab-DUTY replied:
Maybe, it is better to do something like this:
```cmake
if (PROJECT_PLATFORM_E2K)
    target_compile_options($generator_expression$)
endif()
```

Also, I think it is ok have additional target_compile_options for workarounds like this. It makes them more visible and indicates the possibility for cleanup in future, when the bug will probably be fixed.

Masterkatze replied:
Yes, the readability is not the best one :) let's leave it as it is.

r-a-sattarov committed:
xrGame/CMakeLists: added hack for MCST lcc compiler
The MCST lcc compiler does not support a file > 4 Gb, this is an EDG (Edison Design Group) frontend bug, so generation of debug information is disabled

Xottab-DUTY reviewed (approved):
@Masterkatze, thank you!
